### ISSUE TITLE
Fails to build with newer dlist

### ISSUE THREAD

**Issue by relrod** [Deewiant/glob#37]
Although there's a constraint on the `dlist` dep, there's a newer `dlist` out, and `Glob` fails to build against it (e.g. using `--allow-newer`).

```
System/FilePath/Glob/Directory.hs:219:18: error:
    * Couldn't match type `[FilePath]' with `DList FilePath'
      Expected type: IO (DList FilePath, DList FilePath)
        Actual type: IO (DList [Char], [FilePath])
    * In a stmt of a 'do' block:
        return
          $ if null ps then
                (DL.singleton $ DL.head contents ++ replicate n pathSeparator, 
                 DL.tail contents)
            else
                let
                  (matches, nonMatches)
                    = partitionDL fst (fmap (recursiveMatch n m) contents)
                in (fmap snd matches, fmap snd nonMatches)
      In the expression:
        do contents <- getRecursiveContents absPath
           return
             $ if null ps then
                   (DL.singleton $ DL.head contents ++ replicate n pathSeparator, 
                    DL.tail contents)
               else
                   let ... in (fmap snd matches, fmap snd nonMatches)
      In a case alternative:
          True
            | isDir
            -> do contents <- getRecursiveContents absPath
                  return
                    $ if null ps then
                          (DL.singleton $ DL.head contents ++ replicate n pathSeparator, 
                           DL.tail contents)
                      else
                          let ... in ...
    |
219 |                  return $
    |                  ^^^^^^^^...

System/FilePath/Glob/Directory.hs:231:52: error:
    * Couldn't match type `DList' with `[]'
      Expected type: [FilePath]
        Actual type: DList FilePath
    * In the expression: fmap snd nonMatches
      In the expression: (fmap snd matches, fmap snd nonMatches)
      In the expression:
        let
          (matches, nonMatches)
            = partitionDL fst (fmap (recursiveMatch n m) contents)
        in (fmap snd matches, fmap snd nonMatches)
    |
231 |                              in (fmap snd matches, fmap snd nonMatches)
    |                                                    ^^^^^^^^^^^^^^^^^^^

```

**Comment by Deewiant** [comment]
Addressed in e39cc11cedc0e0eaac238d5c46970d436408cdf4, released as v0.10.1: https://hackage.haskell.org/package/Glob-0.10.1

**Comment by relrod** [comment]
@Deewiant Quickest response ever, you're awesome. Thanks so much!
